src/jit/compiler.hpp

```diff
@@ -2843,7 +2843,22 @@ bool                Compiler::fgIsThrowHlpBlk(BasicBlock * block)
     if (!(block->bbFlags & BBF_INTERNAL) || block->bbJumpKind != BBJ_THROW)
         return false;
 
-    GenTreePtr  call = block->bbTreeList->gtStmt.gtStmtExpr;
+    GenTree* call = nullptr;
+    if (block->IsLIR())
+    {
+        // TODO(pdg): it would be nice if there was simply a bit on the block we could check.
+        for (GenTree* node : LIR::AsRange(block))
+        {
+            if (node->OperGet() == GT_CALL)
+            {
+                call = node;
+            }
+        }
+    }
```

Comment:
These are blocks guaranteed to throw, so there will (should) be only a single statement. See fgAddCodeRef()

---
In reply to: [73731740](https://github.com/pgavlin/coreclr/pull/19#discussion_r73731740) [](ancestors = 73731740)